webapp/src/pages/applicationForm/components/ApplicationForm.js

```diff
@@ -276,25 +294,68 @@ class ApplicationForm extends Component {
             currentStep : step - 1
-        })
+        });
     }
 
     handleSubmit = event => {
         event.preventDefault();
         this.setState({
             isLoading: true
         });
-        applicationFormService.submit(this.state.formSpec.id, true, this.state.answers).then(resp=> {
-            let submitError = resp.response_id === null;
-            this.setState({
-                isError: submitError,
-                errorMessage: resp.message,
-                isLoading: false,
-                isSubmitted: resp.is_submitted,
-                submittedTimestamp: resp.submitted_timestamp,
-                responseId: resp.response.id
-              });
-        });
+        if(this.state.new_response){
+            applicationFormService.submit(this.state.formSpec.id, true, this.state.answers).then(resp=> {
+                let submitError = resp.response_id === null;
+                this.setState({
+                    isError: submitError,
+                    errorMessage: resp.message,
+                    isLoading: false,
+                    isSubmitted: resp.is_submitted,
+                    submittedTimestamp: resp.submitted_timestamp,
+                    unsaved_changes : false,
+                    responseId: resp.response.id
+                  });
+            });
+        } else {
+            applicationFormService.updateResponse(this.state.responseId, this.state.formSpec.id, true, this.state.answers).then(resp=> {
+                let saveError = resp.response_id === null;
+                this.setState({
+                    isError: saveError,
+                    errorMessage: resp.message,
+                    isLoading: false,
+                    isSubmitted: resp.is_submitted,
+                    submittedTimestamp: resp.submitted_timestamp,
+                    unsaved_changes : false
+                  });
+            });
+        }
+
+    }
+
+    handleSave = event => {
+        if(this.state.new_response) {
+            applicationFormService.submit(this.state.formSpec.id, false, this.state.answers).then(resp=> {
+                let submitError = resp.response_id === null;
+                this.setState({
+                    isError: submitError,
+                    errorMessage: resp.message,
+                    isSubmitted: resp.is_submitted,
+                    submittedTimestamp: resp.submitted_timestamp,
+                    new_response: false,
+                    unsaved_changes : false
+                  });
+            });
+        } else {
+            applicationFormService.updateResponse(this.state.responseId, this.state.formSpec.id, false, this.state.answers).then(resp=> {
+                let saveError = resp.response_id === null;
+                this.setState({
+                    isError: saveError,
+                    errorMessage: resp.message,
+                    isSubmitted: resp.is_submitted,
+                    unsaved_changes : false
```

Comment:
No change on isLoading or submittedTimestamp? Maybe consider creating a re-usable function that changes the state after update response, might make it easier to not forgot fields.

Reply:
Hey, yeah I didn't update submittedTimestamp because this would just be updating the existing response, and not submitting it (setting is_submitted to true), so there would be nothing to update it to, if that makes sense

Also, why make a change to isLoading? :)

Reply:
I do think you should update submittedTimestamp here, that way the business logic you describe here is all implemented on the server side

Reply:
Regarding isLoading - clicking the save button should set isLoading to true immediately and then set it to false after receiving the response from the server.